### ISSUE TITLE
JSR/RET are not supported with computeFrames option

### ISSUE THREAD

**Issue by xiaoxiaozero** [firebase/firebase-android-sdk#3268]
### [REQUIRED] Step 2: Describe your environment

  * Android Studio version: Arctic Fox | 2020.3.1 Patch 1
  * Firebase Component: perf crash 
  * Component version: gradle 7.2 AGP 7.0.2

** build.gradle(root) 
```
dependencies {
        classpath "com.android.tools.build:gradle:7.0.2"
        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.31"
        classpath "com.google.gms:google-services:4.3.10"
        classpath "com.google.firebase:firebase-crashlytics-gradle:2.8.1"
        classpath "com.google.firebase:perf-plugin:1.4.0"
}
```

** build.gradle(app) 

```
plugins {
    id 'com.android.application'
    id 'kotlin-android'
    id 'com.google.gms.google-services'
    id 'com.google.firebase.firebase-perf'
}

dependencies {
implementation platform('com.google.firebase:firebase-bom:29.0.3')

    //firebase 发版说明 https://firebase.google.com/support/release-notes/android
    // Declare the dependencies for the Crashlytics and Analytics libraries
    // When using the BoM, you don't specify versions in Firebase library dependencies
    implementation 'com.google.firebase:firebase-analytics-ktx'
    implementation 'com.google.firebase:firebase-perf-ktx'
    implementation 'com.google.firebase:firebase-messaging-ktx'

    //远程配置
    implementation 'com.google.firebase:firebase-config-ktx'

    implementation 'com.umeng.umsdk:apm:1.5.2'
    ...
    }
```

### [REQUIRED] Step 3: Describe the problem

```
* What went wrong:
Execution failed for task ':app:mergeExtDexDebug'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Failed to transform apm-1.5.2.aar (com.umeng.umsdk:apm:1.5.2) to match attributes {artifactType=android-dex, asm-transformed-variant=debug, dexing-enable-desugaring=true, dexing-is-debuggable=true, dexing-min-sdk=21, org.gradle.category=library, org.gradle.libraryelements=jar, org.gradle.status=release, org.gradle.usage=java-runtime}.
      > Execution failed for AsmClassesTransform: /Users/xiaoyao/.gradle/caches/transforms-3/a0029789a8d8fddb72a8823b9c5db5cf/transformed/jetified-apm-1.5.2-runtime.jar.
         > JSR/RET are not supported with computeFrames option
```

**Comment by ChinaZeng** [comment]
I also encountered the same problem, please deal with it as soon as possible, thank you.

**Comment by tony-xxw** [comment]
I also encountered the same problem, please deal with it as soon as possible, thank you.

**Comment by argzdev** [comment]
Hi @xiaoxiaozero and @ChinaZeng, thanks for reporting. We'll investigate this and see what we can find.

**Comment by argzdev** [comment]
Hi all, upon investigating it looks like the main issue is the 3rd party library `com.umeng.umsdk:apm:1.5.2`. I tried removing all Firebase related products and still experience the same issue. This may require the update on library of `com.umeng.umsdk:apm` and not on Firebase SDK. With that said, since the problem isn't on the Firebase SDK side, I'll be closing this case. We can re-open this if you have further details to why the issue is pointing to Firebase Perf, or if you have any other questions. Thanks!

**Comment by ChinaZeng** [comment]
> 大家好，经过调查，看起来主要问题是 3rd 方库`com.umeng.umsdk:apm:1.5.2`。我尝试删除所有与 Firebase 相关的产品，但仍然遇到相同的问题。这可能需要更新`com.umeng.umsdk:apm`Firebase SDK 的库而不是 Firebase SDK。话虽如此，由于问题不在 Firebase SDK 方面，我将关闭此案例。如果您对问题指向 Firebase Perf 的原因有进一步的了解，或者您有任何其他问题，我们可以重新打开它。谢谢！

Thank you very much for the quick feedback from the firebase related staff. After my test, the plug-in com.google.firebase.firebase-perf conflicts with the third-party dependency library com.umeng.umsdk:apm:1.5.2. Maybe this is com. google.firebase.firebase-perf handles the problem of not being compatible and perfect.

**Comment by ChinaZeng** [comment]
> Hi all, upon investigating it looks like the main issue is the 3rd party library `com.umeng.umsdk:apm:1.5.2`. I tried removing all Firebase related products and still experience the same issue. This may require the update on library of `com.umeng.umsdk:apm` and not on Firebase SDK. With that said, since the problem isn't on the Firebase SDK side, I'll be closing this case. We can re-open this if you have further details to why the issue is pointing to Firebase Perf, or if you have any other questions. Thanks!

Thank you very much for the quick feedback from the firebase related staff. After my test, the plug-in com.google.firebase.firebase-perf conflicts with the third-party dependency library com.umeng.umsdk:apm:1.5.2. Maybe this is com. google.firebase.firebase-perf handles the problem of not being compatible and perfect.

**Comment by argzdev** [comment]
Hi @ChinaZeng, thank you for the extra details. I was able to reproduce the issue now, and confirmed that `com.umeng.umsdk:apm:1.5.2` does have conflict with `com.google.firebase.firebase-perf`. I will notify an engineer about this. 

Please note that it is currently Christmas season, so a lot of our engineers are off to holidays. They will be able to take a look on this once they get back, perhaps around January. Thank you for your understanding!

**Comment by raymondlam** [comment]
Hi, I was able to reproduce this locally as well. I'm reaching out to the AGP team to see if they can help debug the issue for the AsmClassesTransform.

**Comment by raymondlam** [comment]
Just an update, folks are away over the holidays but we will work with them once they are back.

**Comment by raymondlam** [comment]
The AGP team is looking into the issue with `com.umeng.umsdk:apm`. I'll post an update once they have a fix.

**Comment by raymondlam** [comment]
The AGP team has fixed the issue with the specific library and it will be available with AGP 7.2 beta 2 and AGP 7.3 alpha 2 which should be released next month.
